pkg/metrics/garbage_collector.go

```diff
@@ -0,0 +1,190 @@
+package metrics
+
+import (
+	"reflect"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+	dto "github.com/prometheus/client_model/go"
+	nm "github.com/rancher/norman/metrics"
+	"github.com/rancher/rancher/pkg/settings"
+	"github.com/rancher/types/config"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+var (
+	GCTargetMetricsByNameForClientKey = []interface{}{
+		nm.TotalAddWS, nm.TotalRemoveWS, nm.TotalAddConnectionsForWS, nm.TotalRemoveConnectionsForWS,
+		nm.TotalTransmitBytesOnWS, nm.TotalTransmitErrorBytesOnWS, nm.TotalReceiveBytesOnWS,
+	}
+
+	GCTargetMetricsByIPForPeer = []interface{}{
+		nm.TotalAddPeerAttempt, nm.TotalPeerConnected, nm.TotalPeerDisConnected,
+	}
+)
+
+func MetricGarbageCollection(context *config.ScaledContext) {
```

Comment:
let me create simple struct which will have listers and make `MetricGarbageCollection` instance method of that new struct so that we can define all of listers inside `Register` function and this is similar way to existing controllers